base/applications/utilman/registry.c

```diff
@@ -0,0 +1,201 @@
+
+UTILMAN_GLOBALS Globals;
+REGISTRY_DATA RegData;
+REGISTRY_SETTINGS Settings;
+
+/* DEFINES ********************************************************************/
+
+#define ACCESS_UTILMAN_KEY L"SOFTWARE\\Microsoft\\Windows NT\\CurrentVersion\\Accessibility\\Utility Manager";
+#define UTILMAN_KEY L"SOFTWARE\\Microsoft\\Utility Manager";
+#define OSK_KEY L"On-Screen Keyboard";
+#define MAGNIFIER_KEY L"Magnifier";
+
+/* FUNCTIONS ******************************************************************/
+
+/**
+ * @InitAppRegKey
+ *
+ * Initialize a key. The function may not necessarily create it but open the key
+ * if it already exists. The disposition pointed lpdwDisposition determines that.
+ * This is a function helper.
+ *
+ * @param[in]   hPredefinedKey
+ *     The predefined key (e.g. HKEY_CLASSES_ROOT).
+ *
+ * @param[in]   lpwszSubKey
+ *      The path to the sub key to be created.
+ *
+ * @param[out]   phKey
+ *      A pointer that receives a handle to the key given by the function.
+ *
+ * @param[out]   lpdwDisposition
+ *      A pointer that receives the disposition given by the function.
+ *
+ * @return
+ *      Returns TRUE if the function successfully created a key (or opened it),
+ *      FALSE otherwise for failure.
+ *
+ */
+BOOL InitAppRegKey(IN HKEY hPredefinedKey,
+                   IN LPCWSTR lpwszSubKey,
+                   OUT PHKEY phKey,
+                   OUT LPDWORD lpdwDisposition)
+{
+    LONG lResult;
+
+    lResult = RegCreateKeyExW(hPredefinedKey,
+                              lpwszSubKey,
+                              0,
+                              NULL,
+                              0,
+                              KEY_WRITE,
+                              NULL,
+                              phKey,
+                              lpdwDisposition);
+    if (lResult != ERROR_SUCCESS)
+    {
+        DPRINT("InitAppRegKey(): Failed to create the following key (or open the key) of path \"%S\". The error code is \"%li\".\n", lpwszSubKey, lResult);
+        return FALSE;
+    }
+
+    return TRUE;
+}
+
+/**
+ * @QueryAppSettings
+ *
+ * Query the setting from the application's key. This is a function helper.
+ *
+ * @param[in]   hKey
+ *     A handle to a key.
+ *
+ * @param[in]   lpwszSubKey
+ *      The path to a sub-key.
+ *
+ * @param[in]   lpwszRegValue
+ *      The registry value where we need to get the data from.
+ *
+ * @param[out]   ReturnedData
+ *      An arbitrary pointer that receives the returned data. Being arbitrary,
+ *      the data can be of any type.
+ *
+ * @return
+ *      Returns TRUE if the function successfully loaded the value we wanted,
+ *      FALSE otherwise for failure.
+ *
+ */
+BOOL QueryAppSettings(IN HKEY hKey,
+                      IN LPCWSTR lpwszSubKey,
+                      IN LPCWSTR lpwszRegValue,
+                      OUT PVOID ReturnedData)
+{
+    LONG lResult;
+    HKEY hKeyQueryValue;
+    DWORD cbSize = sizeof(ReturnedData);
```

Comment:
See comment in the other function (sizeof(ReturnedData) will always be == 4 (or 8) in 32-bits (resp. 64-bits) platforms).
That cbSize should be the new extra parameter to the QueryAppSettings function that specifies the maximum size of the buffer pointed by ReturnedData.